packages/integration_test/android/src/main/java/dev/flutter/plugins/integration_test/FlutterDeviceScreenshot.java

```diff
@@ -0,0 +1,174 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package dev.flutter.plugins.integration_test;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.Instrumentation;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Rect;
+import android.os.Build;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.os.Looper;
+import android.view.Choreographer;
+import android.view.PixelCopy;
+import android.view.ViewGroup;
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+import io.flutter.embedding.android.FlutterSurfaceView;
+import io.flutter.embedding.android.FlutterView;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.Result;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+/** FlutterDeviceScreenshot */
+@TargetApi(19)
+class FlutterDeviceScreenshot {
+  /**
+   * Whether the app is run with instrumentation.
+   *
+   * @return true if the app is running with instrumentation.
+   */
+  static boolean hasInstrumentation() {
+    try {
+      return InstrumentationRegistry.getInstrumentation() != null;
+    } catch (IllegalStateException exception) {
+      return false;
+    }
+  }
+
+  /**
+   * Captures a screenshot by drawing the view to a Canvas.
+   *
+   * <p>It also converts {@link FlutterView} to an image view, since {@link FlutterSurfaceView}
```

Comment:
This has some pretty serious performance implications right?

IOW, doing this is incompatible with using integration_test for performance testing.

Is there some other way of doing this without impacting global test state for the rest of the test execution?

Reply:
So the answer seems to be yes. Why would we have this fallback though? Why not just fail if instrumentation is unavailable?

Reply:
Discussed offline: we should probably make this part of the integration_test API so developers can opt into this if they want. If they ahven't opted in, we should fail if instrumentation isn't available.